coalib/settings/ConfigurationGathering.py

```diff
@@ -495,6 +531,9 @@ def gather_configuration(acquire_settings,
         # Note: arg_list can also be []. Hence we cannot use
         # `arg_list = arg_list or default_list`
         arg_list = sys.argv[1:] if arg_list is None else arg_list
+
+    get_style_config(args, config_file='.coafile')
```

Comment:
I think `coala --generate-config` should be an independent command, it should exit after generating the configuration file.